### PR TITLE
feat(#17): implement light mode

### PR DESCRIPTION
**Pull request by tuurep**
Issue: https://github.com/jannis-baum/vivify/issues/17

Okay got this to a great point to start! Raises a few questions:

### No configuration defaults

What should be the default lets say in a case where someone doesn't know what their `prefers-colorscheme` is?

With the changes in this PR's initial commit, I go into the `prefers-colorscheme light` branch, which is coincidentally exactly what I want :smile: However, I don't know what determines this, is it simply the case that if you don't explicitly set `dark`, the default is `light`?

Thinking if there's need for a config option something like `theme`: `[light|dark|system]`, but it's possible that it's not even necessary. What do you think?

### Maintenance of the github syntax theme

Got the syntax theme from here:

https://github.com/highlightjs/highlight.js/blob/main/src/styles/github.css

And as far as I saw, looks great! Looks like someone is "maintainer" of that in the highlight.js repo. Was thinking if since it's like a builtin of highlight.js, is there another way to get it from the lib/module than copypasting the CSS?

Anyways, it has not been updated for 2 years so it doesn't need much attention I think.

### Overflow: auto

I've changed the default of `pre` `overflow: scroll` to `overflow: auto`, because `overflow: scroll` looks pretty harsh on my browser in the case there's nothing to scroll - here's example screenshot:

image

with `auto`, in the above case there's no such thick "borders", but still get the benefit of being able to scroll only in the case it's needed. If there's a reason `overflow: scroll` is needed, let me know :)

### Other

See if the new dark theme blockquote colors work for you, and adjust as necessary.

For the light theme I've stripped out some highly personal choices, mainly with different link-type foreground colors (anchor link and the parent-dir link), so all links are just `a` foreground. Now I'm pretty happy with how this theme would work as a sensible default. Thoughts / criticisms?

### CODE REVIEW

**Comment by jannis-baum** [comment]
Thank you, everything looks great! Also like the block quote in both modes & don't mind about the overflow thing. Sorry for taking a bit longer to review this time.

**tldr;** IMO this is good to merge, I'd just like to include the URL to the GitHub theme in the CSS :)

> Maintenance of the github syntax theme

Cool! I think this is fair, no need for any complicated setup to pull something that's not frequently updated. The katex CSS and LaTeX fonts are also copied straight into this repo. Would you mind adding the source URL into the code as a comment though?

> No configuration defaults

**Disclaimer** - the rest of this is a bit of a rant haha

IMO this is not Vivify's job. The way the CSS is set up now, the dark theme is default and it will change to light iff the user's browser reports `prefers-color-scheme: light`. The way I see it, this is the way it should be [^1]. If every program/app creates their own dark/light config we end up with a huge mess and a pain to switch themes. I think someone who isn't happy with the theme they get on Vivify should take care of their setup themselves, because in this case their issue is not only with Vivify but also with just about every other website that doesn't put (IMO) inappropriate switches into their own config options ^^

The only argument for having a config option that I see is in case someone for whatever reason wants Vivify to have a different theme than everything else, but if this doesn't affect you then I wouldn't add it until someone complains :) Especially because there is also the option to supply your own CSS

[^1]: aside from which theme comes up for (the probably very rare) browsers that don't report anything for `prefers-color-scheme`, but I don't really care about this to be honest🙈

**Review by jannis-baum — APPROVED** [review]

**Comment by tuurep** [comment]
Nice! Fairly surprised I got everything good on the first go :) Added the URL comment.

Thanks for all the work, after this one we've pretty much fixed all of my personal gripes so I can just start using this for writing notes.

I'll probably look into making the AUR package some time this week.

**Comment by jannis-baum** [comment]
> Thanks for all the work, after this one we've pretty much fixed all of my personal gripes so I can just start using this for writing notes.

Awesome! Thank you for all your work as well! Very valuable contributions!

> I'll probably look into making the AUR package some time this week.

Great!
